core/src/Kokkos_Array.hpp

```diff
@@ -320,6 +320,21 @@ struct Array<T, KOKKOS_INVALID_INDEX, Array<>::strided> {
       : m_elem(arg_ptr), m_size(arg_size), m_stride(arg_stride) {}
 };
 
+template <typename T, size_t N>
+KOKKOS_INLINE_FUNCTION constexpr bool operator==(const Array<T, N, void>& lhs,
+                                                 const Array<T, N, void>& rhs) {
+  for (int i = 0; i < (int)(lhs.size()); ++i) {
```

Comment:
can you please just add a comment here to say that the casting is necessary to avoid the pointless comparison warning?

Reply:
Voila voila. I've just added the comment.